shaders/src/surface_depth_main.fs

```diff
@@ -7,7 +7,7 @@ highp vec4 outPicking;
 #       if MATERIAL_FEATURE_LEVEL == 0
 layout(location = 0) out highp vec4 outPicking;
 #       else
-layout(location = 0) out highp vec2 outPicking;
+layout(location = 0) out highp ivec2 outPicking;
```

Comment:
Since you're writing to a RG32UI color texture this should be `uvec2` no? Otherwise Metal complains. This is separate from the missing format issue which I'll look into.

Reply:
done. fixes both vk and metal!